src/components/Skinview.jsx

```diff
@@ -0,0 +1,245 @@
+import React, { useEffect } from 'react';
+import { Container, Row, Col, Button } from 'react-bootstrap';
+import { v4 as uuidv4 } from 'uuid';
+import { saveAs } from 'file-saver';
+import FileInput from './FileInput';
+import JSZip from 'jszip';
+import '../styles/formError.css';
+import '../styles/styles.css';
+
+export default function Skinview() {
+    var skinjson = {
+        skins: [],
+    };
+
+    var imageName = [];
+    var zip = new JSZip();
+
+    // watch inputs for changes (for error handle and user feedback)
+    useEffect(() => {
+        var input = document.querySelectorAll('input');
```

Comment:
Instead of using DOM methods like this the preferred way to do this in react is to use a ref.

https://reactjs.org/docs/refs-and-the-dom.html

Reply:
Dude, thank you so much, you've already helped so much. Ref's seem like a much more elegant way of handling this. Thanks again.